march_rqt_robot_monitor/src/march_rqt_robot_monitor/updater.py

```diff
@@ -1,20 +1,63 @@
 
 import diagnostic_updater
 import rospy
+from sensor_msgs.msg import JointState, Temperature
 from std_msgs.msg import Time
 
-from .diagnostic_analyzers.temperature import temperature_updater
+from .diagnostic_analyzers.control import CheckJointValues
+from .diagnostic_analyzers.temperature import CheckJointTemperature
 from .diagnostic_analyzers.topic_frequency import CheckTopicFrequency
 
 
 def main():
     rospy.init_node('Diagnostic_updater')
+    temperature = False
 
     updater = diagnostic_updater.Updater()
     updater.setHardwareID('MARCH IVc')
 
+    # Frequency checks
     CheckTopicFrequency('Input_Device', '/march/input_device/alive', Time, updater, 5)
```

Comment:
You changed the rest of the diagnostics to expose `diagnostics` to add it to the updater. Except for one, which still accepts the updater. Wouldn't it be nicer to do something like this?
```suggestion
    check_topic_freq = CheckTopicFrequency('Input_Device', '/march/input_device/alive', Time, updater, 5)
    updater.add('Input Device', check_topic_freq.diagnostics)
```

Reply:
I'm thinking about this as well, sadly I haven't thought of a pretty solution.